### PR TITLE
241449 unsupported code lang

### PR DESCRIPTION
**Pull request by adunndevster**
###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/dotnet/docfx/pull/6170)

### CODE REVIEW

**Comment by adunndevster** [comment]
@yufeih - This is ready for review... 

I think one of the tests is broken due to a missing repo asset though: https://dev.azure.com/ceapex/Engineering/_build/results?buildId=165076&view=logs&j=a29a2371-0793-5e14-39b9-20eb0beef0db&t=2dec8569-e360-5b6c-4ee9-b99866856f9d&l=37938

What do we normally do to proceed?

**Comment by yufeih** [comment]
@adunndevster , oops, it was a different test failure: https://dev.azure.com/ceapex/Engineering/_build/results?buildId=165076&view=logs&j=a29a2371-0793-5e14-39b9-20eb0beef0db&t=2dec8569-e360-5b6c-4ee9-b99866856f9d&l=24196

I typically search for `error message` in the logs to find failed test cases. This experience is not pleasant. Maybe I should move test into a dedicated step and upload test results for better display.

File: src/Microsoft.DocAsCode.MarkdigEngine.Extensions/CodeSnippet/HtmlCodeSnippetRenderer.cs [hunk 1]
```diff
@@ -126,26 +135,28 @@ public HtmlCodeSnippetRenderer(MarkdownContext context)
         {
             var result = new Dictionary<string, List<CodeSnippetExtractor>>();
 
-            AddExtractorItems(new[] { "vb", "vbhtml" },
+            AddExtractorItems(new[] { "vb", "vbhtml", "everything" },
```
Comment (yufeih) [inline src/Microsoft.DocAsCode.MarkdigEngine.Extensions/CodeSnippet/HtmlCodeSnippetRenderer.cs:138]: This array here `vb, vbhtml` seems to be serving the same purpose as `LanguageAlias`?

File: src/Microsoft.DocAsCode.MarkdigEngine.Extensions/CodeSnippet/HtmlCodeSnippetRenderer.cs [hunk 2]
```diff
@@ -53,7 +57,8 @@ public class HtmlCodeSnippetRenderer : HtmlObjectRenderer<CodeSnippet>
             { "processing", new List<string>{".pde" } },
             { "python", new List<string>{".py" } },
             { "r", new List<string>{".r" } },
-            { "ruby", new List<string>{"ru", ".ru", ".ruby" } },
+            { "react", new List<string>{".tsx", "tsx" } },
```
[the remark this answers is not in the document]
Comment (adunndevster) [reply src/Microsoft.DocAsCode.MarkdigEngine.Extensions/CodeSnippet/HtmlCodeSnippetRenderer.cs:60]: Good catch. We may not even need to do this at all, let me look.

File: src/Microsoft.DocAsCode.MarkdigEngine.Extensions/CodeSnippet/HtmlCodeSnippetRenderer.cs (hunk 1 above)
Comment (adunndevster) [reply src/Microsoft.DocAsCode.MarkdigEngine.Extensions/CodeSnippet/HtmlCodeSnippetRenderer.cs:138]: vb and vbhtml are just references to the `LanguageAlias` dictionary where you will find vb related aliases such as "vbnet", "vbscript", ".vb", ".bas", ".vbs", ".vba", etc.

File: src/Microsoft.DocAsCode.MarkdigEngine.Extensions/CodeSnippet/HtmlCodeSnippetRenderer.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (adunndevster) [reply src/Microsoft.DocAsCode.MarkdigEngine.Extensions/CodeSnippet/HtmlCodeSnippetRenderer.cs]: Okay, refactored this.

**Comment by adunndevster** [comment]
@yufeih - Whew! All the unit tests passed. I looked through the regression tests and they look good to me. It's amazing how such a small change can make so many diffs.

In the diffs, there was lots of...
```
<p class=\"mx-imgBorder\">⬇
+<p><span class=\"mx-imgBorder\">⬇
 <img src=\"../media/waf-front-door-configure-ip-restriction/custom-rule.png\" alt=\"Custom rule\" data-linktype=\"relative-path\">⬇
+</span>⬇
```

These are all harmless and even necessary. This change makes it so that images can be rendered inline with text.

Another diff that stuck out to me was in the DevSandbox. There is a new warning I added, "It looks like your code snippet was not rendered. Try range instead." I looked at each of these cases, and this is a valid warning, that correctly exposes instances where either the id was not referenced correctly, or the code snippet file was not setup correctly.

Let me know if you see anything concerning in the diffs, otherwise I think this is ready.

**Comment by yufeih** [comment]
LGTM , there are a couple of tiny performance tweaks that I can make after this PR gets merged.

**Pull request merged**